sdk/eventhub/azure-eventhub/tests/livetest/asynctests/test_send_async.py

```diff
@@ -155,15 +156,38 @@ async def test_send_with_partition_key_async(connstr_receivers):
         await client.send_batch(await client.create_batch())
 
     found_partition_keys = {}
+    reconnect_receivers = []
     for index, partition in enumerate(receivers):
-        received = partition.receive_message_batch(timeout=5000)
-        for message in received:
+        retry_total = 0
+        while retry_total < 3:
+            timeout = 5000 + retry_total * 1000
             try:
-                event_data = EventData._from_message(message)
-                existing = found_partition_keys[event_data.partition_key]
-                assert existing == index
-            except KeyError:
-                found_partition_keys[event_data.partition_key] = index
+                received = partition.receive_message_batch(timeout=timeout)
+                for message in received:
+                    try:
+                        event_data = EventData._from_message(message)
+                        existing = found_partition_keys[event_data.partition_key]
+                        assert existing == index
+                    except KeyError:
+                        found_partition_keys[event_data.partition_key] = index
+                if received:
+                    break
+                retry_total += 1
+            except uamqp.errors.ConnectionClose:
```

Comment:
is partition receiver (ReceiveClient) defined in pytest fixture shared by multiple test cases? or for each test case it would be a new partition receiver obj

Reply:
It's shared by multiple test cases by default, unless we were to set the `fixture(scope="session")` or some other scope that isn't every test.